src/main/java/com/google/plantasy/utils/GameUtils.java

```diff
@@ -42,32 +43,31 @@ public static Entity createGameEntity(String gameName, DatastoreService datastor
     
     // intializing game property values
     Entity gameEntity = new Entity("Game");
+    Game game = new Game(gameEntity);
     ArrayList<String> userIds = new ArrayList<String>();
     String quizQuestion = "";
     long quiz_timestamp = 0;
- 
-    gameEntity.setProperty("gameName", gameName);
-    gameEntity.setProperty("userIds", userIds);
-    gameEntity.setProperty("quizQuestion", quizQuestion);
-    gameEntity.setProperty("quiz_timestamp", quiz_timestamp);
     Key gameKey = gameEntity.getKey();
- 
+
+    game.setGameName(gameName);
+    game.setUserIds(userIds);
+    game.setQuizQuestion(quizQuestion);
+    game.setQuizTimestamp(quiz_timestamp);
+
     datastore.put(gameEntity);
 
     // getting the game id for the game id property
-    Entity entity = null;
     try{
-      entity = datastore.get(gameKey);
-
-      String key = KeyFactory.keyToString(gameKey);
-      entity.setProperty("gameId", key);
-      datastore.put(entity);
+      gameEntity = datastore.get(gameKey);
```

Comment:
I'm a bit confused... why do we need to do this? It looks like we only get the entity just to put it back into the datastore on line 68. Do we need to get it at all?

Reply:
You're right, I took that out.